### ISSUE TITLE
No module named '_sysconfigdata__darwin_x86_64-apple-darwin22'

### ISSUE THREAD

**Issue by 13am** [pyenv/pyenv#2674]
Installing pythons 3.9, 3.10, 3.11 work.  However, anything below 3.9 fails with

`ModuleNotFoundError: No module named '_sysconfigdata__darwin_x86_64-apple-darwin22'`

### Description
- [x] Platform information: MacOS Ventura 13.3
- [x] OS architecture: amd64
- [x] pyenv version: pyenv 2.3.17
- [x] Python version: 3.8 and below
- [x] C Compiler information (e.g. gcc 7.3): 
```
clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: x86_64-apple-darwin22.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

```
- [x] Please attach the debug trace of the failing command as a gist:
  * Run `env PYENV_DEBUG=1 <faulty command> 2>&1 | tee trace.log` and attach `trace.log`. E.g. if you have a problem with installing Python, run `env PYENV_DEBUG=1 pyenv install -v <version> ` (note the `-v` option to `pyenv install`).
  [trace.log](https://github.com/pyenv/pyenv/files/11171425/trace.log)

- [ ] If you have a problem with installing Python, please also attach `config.log` from the build directory
  * The build directory is reported after the "BUILD FAILED" message and is usually under `/tmp`.
- [ ] If the build succeeds but the problem is still with the build process (e.g. the resulting Python is missing a feature), please attach
  * the debug trace from reinstalling the faulty version with `env PYENV_DEBUG=1 pyenv install -f -k -v <version> 2>&1 | tee trace.log`
  * `config.log` from the build directory. When using `pyenv install` with `-k` as per above, the build directory will be under `$PYENV_ROOT/sources`.

<- I did not find those, but I'll attach the build log, which was mentioned in trace.log:

[python-build.20230406170752.70972.log](https://github.com/pyenv/pyenv/files/11171441/python-build.20230406170752.70972.log)

**Comment by native-api** [comment]
Most likely, you have Homebrew `binutils` and/or `llvm` on `PATH`: here, the name of that module is `_sysconfig__darwin_darwin`.

**Comment by 13am** [comment]
Thank you for your quick response! I did not have any of those in my PATH. However, as you implied this may have something to do with packages installed by Homebrew, I started blindly uninstalling/installing every package indicated in the build output as  "python-build: use [package] from homebrew". Magically, after

```
brew uninstall tcl-tk
brew install tcl-tk
```

builds started working! Thank you for pointing me to the right direction.
